### DISCUSSION TITLE
Cannot get any of the images to install on my pi 4 with touchscreen and jabra speaker/mic combo

### DISCUSSION THREAD

**Discussion by Steve973** [OpenVoiceOS/ovos-buildroot#121] [Q&A]
Hello.  I spent a good deal of time trying to get OVOS installed on my pi 4, but I was not successful.  The same setup works immediately with Mycroft (picroft).  I tried each of the images at http://downloads.openvoiceos.com/images/ with nearly identical results from each.  

The devkit image booted to a screen that gives the option to perform setup with a phone, or to perform on the local device, with a "skip" option beneath.  The "skip" option is highlighted, and I cannot select either of the other two options with touchscreen, keyboard, or mouse.  I can only press the space bar, which selects the highlighted "skip" option, after which it goes to the animated logo screen and I cannot do anything else.

The other images all do exactly the same thing: they bring up a screen instructing me to connect to the OVOS wifi SSID, and a second later, it flashes a red "connection failed" screen, and it loops between the two.  I can sometimes see the OVOS SSID with my phone, but there is not enough time for me to do anything before it kills it and does it all over again.

Any ideas?  It is my understanding that it should work with my hardware, but I don't have a lot of options.  I haven't tried bringing up another terminal to log in and perform any diagnostics.

**Comment by AIIX** [comment]
Would help to know the screen information, it seems the touch screen is not recognized  at all

**Reply by Steve973** [reply]
It's a SunFounder 10.1 IPS LCD monitor.  I misspoke, and it is not a touchscreen.  Either way, isn't there another way to select whether you want to do the setup with a phone, or locally?  Is a touchscreen required?  I can still return the screen that I have and get a touchscreen if it's mandatory.  Then, with the image that kept trying to create the hotspot and reporting a failure immediately, repeating over and over, why might that be happening?  Does it matter which image I use with the pi 4?

**Reply by AIIX** [reply]
The images are currently optimized for a smartspeaker type setup with a touch screen, will also need a touch screen to navigate most of the UI, on mobile wifi setup is a broken on current manjaro development images is why you get the broken loop, a fix is in the works for that image base

If you do want to use a mouse instead of the touchscreen you will need to manually edit the following file: /usr/lib/systemd/system/mycroft-gui.service and remove the hide cursor environment variable so the cursor is not hidden on boot in the image

**Comment by Steve973** [comment]
Thank you.  What is the definition of a smart speaker?  I have the Jabra
510.

**Reply by AIIX** [reply]
There is no specific definition but in general what any smartspeaker with a screen has, the current manjaro image is optimized for this:

1. Base sbc: rpi4
2. Touchscreen for ui input: any supported by rpi4 or linux
3. Speakers for speech output: currently image outputs to HDMI, can be configured via pulse for another output
4. Mic for speech input: currently configured for respeaker mic array, can be reconfigured for another mic input via pulse

**Comment by Steve973** [comment]
Ok. Thanks.  I'll update to a touchscreen in a couple of days.  I'll see if
that gets me past configuration, and then I can work on pulse audio.

**Comment by Steve973** [comment]
Ok, update on my experience:  I switched to a touchscreen and I got through the installation.  It's a little odd that leaving the touchscreen USB connector connected while I boot causes the Pi 4 to be unable to boot.  But I can unplug it, boot, and plug it back in, and the touchscreen works fine.  During setup, when I tried to choose the online backend, it informed me that there was an error, so I had to choose the offline backend to continue.  The home screen shows up, where I get the time and the date, but I have no idea what the wake word is (tried "hey mycroft" to no avail) but I am not even sure that the mic is working with OVOS.

So, with my Pi and my Jabra USB speakerphone/mic combo, which image would be best for me to run?  And, what do you recommend that I do next?  Do I need to SSH into it in order to configure pulse?  It seems to be able to use the USB speaker right away, but maybe the mic issue is that I need to reconfigure it in pulse.

**Reply by AIIX** [reply]
There is a new image available based on fkms and with newer fixes, try that and yes you will need to configure pulse to use your mic as it's configured for the respeaker by default

**Comment by Steve973** [comment]
Thank you.  I will try it immediately.  Is there anything that you would like me to try while I am testing this?

**Reply by AIIX** [reply]
Username is mycroft and pass is 12345 for the image, ssh is enabled by default

**Comment by Steve973** [comment]
The installation went perfectly.  Somehow, it is using the mic from the Jabra unit, but it is using the speakers built into the touchscreen enclosure.  I haven't worked much with pulse.  I am looking at /etc/pulse/default.pa, and /etc/pulse/system.pa though I am not sure where it's configured to use the respeaker.

**Reply by AIIX** [reply]
You can try using pactl:

List mics available on system:
pactl list sources short
pactl set-default-source *the-input-name*

List available speakers:
pactl list sinks short
pactl set-default-sink *the-input-name*

**Reply by Steve973** [reply]
Thanks.  It's working and I didn't have to do much of that.  It doesn't seem to know how to do anything.  I even tell it the things that it suggests on the home screen, and it doesn't know what they mean.  Is it a lack of installed skills, or something that hasn't been set up?  I thought that I could tell it, "Hey, Mycroft, install &lt;skill name&gt;" but that doesn't work, either.  Is that all done through the command line?

**Reply by Steve973** [reply]
I'm figuring some of this out on my own, finally.  This image is working quite well, I think!
